.github/workflows/release.yml

```
@@ -0,0 +1,11 @@
+name: Publish Release
+
+on:
+  release:
+    types: [published]
+    branches: [main]
```

this project has name `master` for main branch
```suggestion
    branches: [master]
```